frontend/src/app/shared/components/searchable-project-list/searchable-project-list.service.ts

```diff
@@ -223,4 +300,64 @@ export class SearchableProjectListService {
     (listParent?.querySelector('.spot-list--item-action_active') as HTMLElement)?.click();
     (focused as HTMLElement)?.focus();
   }
+
+  /**
+   * Fetches projects according to extraIds and appends them to projects. Fetching will only be performed for IDs that
+   * are not already present in projects.
+   * @param {IProject[]} projects - The initial projects that new fetches will be appended to
+   * @param {string[]} concatIds - A list of project IDs that identifies projects that shall be appended
+   */
+  private pipeConcatProjects(projects:IProject[], concatIds:string[], extraFetches:Observable<IHALCollection<IProject>>[] = []) {
+    const existingIds = this.extractIds(projects);
+    concatIds = concatIds.filter((id) => !existingIds.has(id));
+
+    for(let sliceStart = 0; sliceStart < concatIds.length; sliceStart += this.maximumPageSize) {
+      const extraFilter = [['id', '=', concatIds.slice(sliceStart, sliceStart + this.maximumPageSize)]] as ApiV3ListFilter[];
+      extraFetches.push(
+        this.fetchProjects(extraFilter, this.maximumPageSize)
+      );
+    }
+
+    if(extraFetches.length === 0) {
+      return of(projects);
+    }
+
+    return forkJoin(extraFetches).pipe(
+      map((collections) => collections.map((collection) => collection._embedded.elements)),
+      map((collections) => projects.concat(...collections)),
+      map((allProjects) => _.uniqBy(allProjects, (p) => p.id)),
+    );
+  }
+
+  private fetchProjects(filters:ApiV3ListFilter[] = [], pageSize?:number):Observable<IHALCollection<IProject>> {
+    const query = listParamsString(this.params(filters, pageSize));
+    return this.http.get<IHALCollection<IProject>>(this.apiV3Service.projects.path + query);
+  }
+
+  private extractIds(projects:IProject[]):Set<string> {
+    return new Set<string>(projects.map((p) => p.id.toString()));
+  }
+
+  private extractAncestors(projects:IProject[]):string[] {
+    const ancestors = new Set<string>();
+    projects.forEach((p) => p._links.ancestors.forEach((a) => ancestors.add(a.href)));
+
+    // FIXME: Why can't I just call values().filter(...)? According to MDN this is recommended
```

Comment:
❓ Do you want to leave those FIXMEs in here for now? Did you actually talked to Alex Coles about the ts target?

Reply:
Yeah I actually wanted to leave those. As in: Apparently we are not targeting a new enough ES and I didn't want to get into that topic. Maybe I should've rephrased the whole comment to be more like "I know why this is not the case..."